### PR TITLE
MIT-licensed sparse() parent method and expert driver, take two

### PR DESCRIPTION
**Pull request by Sacha0**
See #14798 (the original pull request) for discussion. Apologies for the close/reopen-glitch!

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -91,6 +91,8 @@ Library improvements
     * Rank one update and downdate functions, `lowrankupdate`, `lowrankupdate!`, `lowrankdowndate`,
     and `lowrankdowndate!`, for dense Cholesky factorizations ([#14243],[#14424])
 
+    * All `sparse` methods now retain provided numerical zeros as structural nonzeros.
```
Comment (tkelman) [inline NEWS.md]: mention `SparseArrays.dropzeros!` as the way to get rid of them when the user would like to do so?

Comment (Sacha0) [inline NEWS.md]: Sounds wise. In fact, given `dropzeros!` should become increasingly important / regularly used with the shift to retaining numerical zeros by default, perhaps `dropzeros!` should now be exported?

Comment (StefanKarpinski) [inline NEWS.md]: +1 for exporting `dropzeros!` and making other operations retain zeros (including the `sparse` function).

**Comment by Sacha0** [comment]
> mention SparseArrays.dropzeros! as the way to get rid of them when the user would like to do so?

I added lines to this effect to `sparse`'s documentation and NEWS.md (without the `SparseArrays` qualification, anticipating export of `dropzeros!`).

Assuming exporting `dropzeros!` seems advantageous to all thread participants, should I add a separate commit to this PR exporting `dropzeros!`? Or would opening a separate PR (to broaden the change's visibility and hence opportunity for comment) be better?

**Comment by tkelman** [comment]
Maybe do the export separately, I'd rather get this merged. Hopefully @jrevels and others will figure out the right way to standardize running package-based benchmarks over time.

When it does get exported, it should also be added to the docs and probably tested a bit more extensively.

**Comment by Sacha0** [comment]
> Maybe do the export separately, I'd rather get this merged. Hopefully @jrevels and others will figure out the right way to standardize running package-based benchmarks over time.
> 
> When it does get exported, it should also be added to the docs and probably tested a bit more extensively.

Sounds great. Let me know if any other touch-ups are in order here. Thanks again, and best!

**Comment by tkelman** [comment]
apt-get travis issue addressed by #15249, restarted
```
dpkg: dependency problems prevent configuration of gfortran-5:i386:
 gfortran-5:i386 depends on gcc-5 (= 5.2.1-23ubuntu1~12.04); however:
  Package gcc-5:i386 is not configured yet.
dpkg: error processing gfortran-5:i386 (--configure):
 dependency problems - leaving unconfigured
Setting up libdpkg-perl (1.16.1.2ubuntu7.7) ...
Setting up zlib1g-dev:i386 (1:1.2.3.4.dfsg-3ubuntu4) ...
Setting up libssl-dev:i386 (1.0.1-4ubuntu5.34) ...
Setting up make:i386 (3.81-8.1ubuntu1.1) ...
Processing triggers for libc-bin ...
ldconfig deferred processing now taking place
Errors were encountered while processing:
 gcc-5:i386
 g++-5:i386
 gfortran-5:i386
E: Sub-process /usr/bin/dpkg returned an error code (1)
```

**Comment by Sacha0** [comment]
@tkelman Much thanks for the thorough review!

**Pull request merged**